model/infinispan/src/main/java/org/keycloak/models/sessions/infinispan/util/SessionTimeouts.java

```diff
@@ -114,7 +114,7 @@ public static long getClientSessionLifespanMs(RealmModel realm, AuthenticatedCli
 
```

Comment:
@pjbober Wait a second, I think there is still a mistake here. It can happen that when using the `getClientSessionMaxLifespan` the substraction ends with a negative result. Let me think out loud. Image the user session was started at 0 but client session was started later at 60, session lifespan is 100 but client lifespan is 50. The current time is 70. This method now returns:

```
timeSinceUserSessionStart = 60 - 0 = 60
sessionMaxLifespan = realm.getClientSessionMaxLifespan() = 50;
timeToExpire = sessionMaxLifespan - timeSinceUserSessionStart = 50 - 60 = -10;
```

I think we have to re-think this. I'll try to get back to you.